src/components/pages/data/table/expanded.tsx

```diff
@@ -0,0 +1,123 @@
+import { SimpleGrid, Table, TableContainer, Tbody, Td, Tr } from "@chakra-ui/react";
+import { Prose } from "@nikolovlazar/chakra-ui-prose";
+import { axGetCCAData } from "@services/cca.service";
+import { formatDate } from "@utils/date";
+import { getInjectableHTML } from "@utils/text";
+import React, { useEffect, useMemo, useState } from "react";
+
+export default function ExpandedComponent(props) {
+  const ccaId = props.data.id;
+  const [data, setData] = useState({});
+
+  const ccaData = async () => {
+    const { success, data } = await axGetCCAData(ccaId);
+    if (success) {
+      setData(data);
+    }
+  };
+
+  useEffect(() => {
+    ccaData();
+  }, []);
+
+  const EXPAND_BLACKLIST = ["id", "ccaFieldValues"];
+
+  const infoRows = useMemo(
+    () => Object.entries(data).filter((r) => r[1] && !EXPAND_BLACKLIST.includes(r[0])),
+    [data]
+  );
+
+  function findCCAFieldValuesByKey(obj, keyToFind) {
+    return (
+      Object.entries(obj).reduce(
+        (acc, [key, value]) =>
+          key === keyToFind
+            ? acc.concat(obj)
+            : typeof value === "object" && value
+              ? acc.concat(findCCAFieldValuesByKey(value, keyToFind))
+              : acc,
+        []
+      ) || []
+    );
+  }
+
+  const buildObject = (arr) => {
+    const obj = {};
+    for (let i = 0; i < arr.length; i++) {
```

Comment:
can you refactor this?\
suggestion : extract the if/else conditional logic in a separate function and use map() function to iterate through the array. avoid using for loops directly